device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py

```diff
@@ -69,101 +66,91 @@ def get_log_ports(handle, sfp_module):
+
+    if cmd == "enable":
+        pwr_attr_set(handle, module_id, log_port_list, SX_MGMT_PHY_MOD_PWR_ATTR_PWR_MODE_E, SX_MGMT_PHY_MOD_PWR_MODE_LOW_E)
+        print "Enabled low power mode for module [%d]" % (module_id)
+    elif cmd == "disable":
+        pwr_attr_set(handle, module_id, log_port_list, SX_MGMT_PHY_MOD_PWR_ATTR_PWR_MODE_E, SX_MGMT_PHY_MOD_PWR_MODE_AUTO_E)
+        print "Disabled low power mode for module [%d]" % (module_id)
     else:
-        pmaos.admin_status = PMAOS_ENABLE
-
-    meta.access_cmd = SXD_ACCESS_CMD_SET
-    rc = sxd_access_reg_pmaos(pmaos, meta, REGISTER_NUM, None, None)
-    assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_pmaos failed, rc = %d" % rc
-
-def set_sfp_lpmode(sfp_module, lpm_enable):
-    # Get PMMP
-    pmmp = ku_pmmp_reg()
-    pmmp.module = sfp_module
-    meta = init_sx_meta_data()
-    meta.access_cmd = SXD_ACCESS_CMD_GET
-    rc = sxd_access_reg_pmmp(pmmp, meta, REGISTER_NUM, None, None)
-    assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_pmmp failed, rc = %d" % rc
-
-    # Set low power mode status
-    lpm_mask = 1 << 8
-    if lpm_enable:
-        pmmp.eeprom_override = pmmp.eeprom_override | lpm_mask
-    else:
-        pmmp.eeprom_override = pmmp.eeprom_override & (~lpm_mask)
-
-    meta.access_cmd = SXD_ACCESS_CMD_SET
-    rc = sxd_access_reg_pmmp(pmmp, meta, REGISTER_NUM, None, None)
-    assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_pmmp failed, rc = %d" % rc
+        print("Error: Invalid command")
+        sys.exit(0)
 
-# Check if SFP port number is provided
 if len(sys.argv) < 3:
     print "SFP module number or LPM is missed."
     print "Usage: sfplpmset.py <SFP module> <on|off>"
     sys.exit(errno.EINVAL)
 
+cmd = None
 lpm_enable = None
 if sys.argv[2] == 'on':
     lpm_enable = True
+    cmd = 'enable'
 elif sys.argv[2] == 'off':
     lpm_enable = False
+    cmd = 'disable'
 else:
     print "Unrecognized LPM parameter. Please use <on> or <off> values"
     sys.exit(errno.EINVAL)
 
-# Init SDK API
-rc, handle = sx_api_open(None)
-if (rc != SX_STATUS_SUCCESS):
-    print "Failed to open api handle.\nPlease check that SDK is running."
-    sys.exit(errno.EACCES)
-
-pid = os.getpid()
-rc = sxd_access_reg_init(pid, None, SXD_LOG_VERBOSITY_LEVEL)
-if (rc != SXD_STATUS_SUCCESS):
-    print "Failed to initializing register access.\nPlease check that SDK is running."
-    sys.exit(errno.EACCES);
-
 # Get SFP module
 sfp_module = int(sys.argv[1]) - 1
 
-# Get all ports at admin up status that related to the SFP module
-log_port_list = get_log_ports(handle, sfp_module)
-
-# SET SFP related ports to admin down status
-for log_port in log_port_list:
-    set_port_admin_status_by_log_port(handle, log_port, SX_PORT_ADMIN_STATUS_DOWN)
-
-# Disable admin status before LPM settings
-set_sfp_admin_status(sfp_module, SX_PORT_ADMIN_STATUS_DOWN)
+print "[+] opening sdk"
+rc, handle = sx_api_open(None)
+print ("sx_api_open handle:0x%x , rc %d " % (handle, rc))
```

Comment:
Not sure if this print is for debugging or not. If it is for debuggging, we might need remove it.

Reply:
fixed